packages/lib/src/vis/shared/PanZoomMesh.tsx

```diff
@@ -103,8 +108,17 @@ function PanZoomMesh(props: Props) {
     (evt: ThreeEvent<WheelEvent>) => {
       const { sourceEvent, unprojectedPoint } = evt;
 
+      if (!zoom) {
+        return;
+      }
+
       const factor = sourceEvent.deltaY > 0 ? ZOOM_FACTOR : 1 / ZOOM_FACTOR;
-      const zoomVector = new Vector3(1 / factor, 1 / factor, 1);
+      const noKeyPressed = noModifierKeyPressed(sourceEvent);
+      const zoomVector = new Vector3(
+        noKeyPressed || isEventValid(sourceEvent, zoomKeyX) ? 1 / factor : 1,
+        noKeyPressed || isEventValid(sourceEvent, zoomKeyY) ? 1 / factor : 1,
+        1
```

Comment:
As discussed, a WheelEvent with no key pressed should always allow to zoom/unzoom in both directions. If a key a pressed, I check if this corresponds to the X zoom or the Y zoom to set the `zoomVector` values.

Reply:
That being said, is there a point in preventing zoom when a modifier key is pressed and `x/yZoom` is `false`? It may be convenient to be able to zoom while drawing a selection? 

In this case, the ternary could be simplified and inverted: `!xZoom || (xZoom && isEventValid(sourceEvent, xZoomKey)) ? 1 : 1 / factor` and since `xZoomKey` is always defined, this could just become `!xZoom || (xZoom && sourceEvent.getModifierState(xZoomKey)) ? 1 : 1 / factor`.

Since `isEventValid` and `noModifierKeyPressed` would no longer be used for the zoom, you'd then be able to inline the `isEventValid` logic in the pan callback: 

```ts
if (!pan) {
  return;
}

const isPanAllowed = panKey ? event.getModifierState(panKey) : noModifierKeyPressed(event);
if (isPanAllowed) {
  (target as Element).setPointerCapture(pointerId); // https://stackoverflow.com/q/28900077/758806
  startOffsetPosition.current = unprojectedPoint.clone();
}
```

Reply:
As discussed, breaking down the pan/zoom in multiple components should allow to solve such quandaries.

Before that, removing `isEventValid` is appealing to me so I will implement your proposed changes, merge as it is and work on the breaking down next.

Reply:
Yep, no worries 👍